spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -0,0 +1,571 @@
+/// requests or notifications to the server.
+@jsonRequest("build/initialize")
+operation InitializeBuild {
+    input: InitializeBuildParams
+    output: InitializeBuildResult
+}
+
+@jsonNotification("build/initialized")
+operation OnBuildInitialized {
+    input: InitializeBuildParams
+}
+
+/// Like the language server protocol, the shutdown build request is sent from the client to the server. It asks the server to shut down,
+/// but to not exit (otherwise the response might not be delivered correctly to the client). There is a separate exit notification that
+/// asks the server to exit.
+@jsonRequest("build/shutdown")
+operation ShutdownBuild {
+
+}
+
+@jsonNotification("build/exit")
+/// Like the language server protocol, a notification to ask the server to exit its process. The server should exit with success code 0
+/// if the shutdown request has been received before; otherwise with error code 1.
+operation OnBuildShutdown {
+
+}
+
+@jsonRPC
+service BuildClient {
+    operations: [
+        ShowMessage
+        LogMessage
+        PublishDiagnostics
+    ]
+}
+
+/// The show message notification is sent from a server to a client to ask the client to display a particular message in the user interface.
+///
+/// A build/showMessage notification is similar to LSP's window/showMessage, except for a few additions like id and originId.
+@jsonNotification("build/showMessage")
+operation ShowMessage {
+    input: ShowMessageParams
+}
+
+/// The log message notification is sent from a server to a client to ask the client to log a particular message in its console.
+///
+/// A build/logMessage notification is similar to LSP's window/logMessage, except for a few additions like id and originId.
+@jsonNotification("build/logMessage")
+operation LogMessage {
+    input: LogMessageParams
+}
+
+/// The Diagnostics notification are sent from the server to the client to signal results of validation runs.
+///
+/// Diagnostic is defined as it is in the LSP
+///
+/// When reset is true, the client must clean all previous diagnostics associated with the same textDocument and
+/// buildTarget and set instead the diagnostics in the request. This is the same behaviour as PublishDiagnosticsParams
+/// in the LSP. When reset is false, the diagnostics are added to the last active diagnostics, allowing build tools to
+/// stream diagnostics to the client.
+///
+/// It is the server's responsibility to manage the lifetime of the diagnostics by using the appropriate value in the reset field.
+/// Clients generate new diagnostics by calling any BSP endpoint that triggers a buildTarget/compile, such as buildTarget/compile, buildTarget/test and buildTarget/run.
+@jsonNotification("build/publishDiagnostics")
+operation PublishDiagnostics {
+    input: PublishDiagnosticsParams
+}
+
+/// A resource identifier that is a valid URI according
+/// to rfc3986: * https://tools.ietf.org/html/rfc3986
+string URI
```

Comment:
You can define "type aliases" for primitive types, as such. It makes the spec easier to understand imho.